src/Serilog/Settings/KeyValuePairs/SettingValueConversions.cs

```diff
@@ -29,6 +35,22 @@ class SettingValueConversions
 
         public static object ConvertToType(string value, Type toType)
         {
+            if (TryParseStaticMemberAccessor(value, out var accessorTypeName, out var memberName))
```

Comment:
Perhaps this could move down to the `if ((toTypeInfo.IsInterface || toTypeInfo.IsAbstract) && !string.IsNullOrWhiteSpace(value))`  block? If it's first up, here, we might hit issues where this is ambiguous, e.g. the target `Type` is `string`.

Reply:
That was my initial implementation, but then I thought it would be strange to limit it to only `abstract class`es and `interface`s, because after all this can also be useful in other cases, like for sinks that accept *complex* parameters that are not `abstract` nor `interface` (for instance : `EmailConnectionInfo` in *Serilog.Sinks.Email* , `ColumnOptions` for *Serilog.Sinks.MSSqlServer* ...) 

This also gives a workaround for configuration that canot be easily defined with the keay-value settings syntax, by allowing to define it as code in a static property.

Also, I've tried to make the *regex* for static-property accessor as specific as possible to avoid false positives.  

What do you think ?

Reply:
@nblumhardt any opinion on the subject? :)

Reply:
I think it's a fair consideration - just not sure if it's behavior we want to commit to yet in this PR. (We'd also then need to consider whether assembly-qualified type names should be supported in more places, since subclassing is a possible workaround for other scenarios, plus look at how this would interact with some kind of "structure" syntax for populating fields of complex configuration parameters down the track).

TL;DR: I don't think it's a bad idea - just might be scope creep for this PR, since we haven't really dug through all the possible implications :-) .. what do you think?

Reply:
That makes sense 👍 
Easier to make it support a few use cases initially and extend it later on if needed.

Thanks !